Improve document for IDE support and simplify doc build (#654)
apache/spark-website: README.md

In this directory you will find text files formatted using Markdown, with an `.md` suffix.

Building the site requires [Ruby 3](https://www.ruby-lang.org), [Jekyll](http://jekyllrb.com/docs), and
[Rouge](https://github.com/rouge-ruby/rouge). The most reliable way to ensure a compatible environment
is to use the official Docker build image from the Apache Spark repository.

If you haven't already, clone the [Apache Spark](https://github.com/apache/spark) repository. Navigate to
the Spark root directory and run the following command to create the builder image:
```
docker build \
  --tag docs-builder:latest \
  --file dev/spark-test-image/docs/Dockerfile \
  dev/spark-test-image-util/docs/
```

Once the image is built, navigate to the `spark-website` root directory, run the script which processes
the Markdown files in the Docker container.
```

> SPARK_WEBSITE_PATH="/path/to/spark-website" sh .dev/build-docs.sh

Reviewer: I think env var `SPARK_WEBSITE_PATH` is not used. and grant `x` permission to `.dev/*.sh` then we don't need use `sh foo.sh` to run the script

Reviewer: opened https://github.com/apache/spark-website/pull/675 to address the issues I found

Author: SGTM

In this directory you will find text files formatted using Markdown, with an `.md` suffix.

Building the site requires [Ruby 3](https://www.ruby-lang.org), [Jekyll](http://jekyllrb.com/docs), and
[Rouge](https://github.com/rouge-ruby/rouge). The most reliable way to ensure a compatible environment
is to use the official Docker build image from the Apache Spark repository.

If you haven't already, clone the [Apache Spark](https://github.com/apache/spark) repository. Navigate to
the Spark root directory and run the following command to create the builder image:
```
docker build \
  --tag docs-builder:latest \
  --file dev/spark-test-image/docs/Dockerfile \
  dev/spark-test-image-util/docs/
```

> Once the image is built, run the container to process the Markdown files. Note: Replace `/path/to/spark-website`

Author: ```suggestion
Once the image is built, navigate to the spark-website root directory, run the script
to process the Markdown files. Note: Replace `/path/to/spark-website`
```

in the command below with the *absolute path* to your local website directory.
```
docker run \
  -e HOST_UID=$(id -u) \
  -e HOST_GID=$(id -g) \
  --mount type=bind,source="/path/to/spark-website",target="/spark-website" \
  -w /spark-website \
  docs-builder:latest \

>   /bin/bash -c "sh .dev/run-in-container.sh"

Author: ```suggestion
SPARK_WEBSITE_PATH="/path/to/spark-website" sh .dev/build-docs.sh
```

>   /bin/bash -c "sh .github/run-in-container.sh"

Reviewer 2: `.github`? Maybe it is better to be in `dev`?

>   /bin/bash -c "sh .github/run-in-container.sh"

Author: ```suggestion
  /bin/bash -c "sh .dev/run-in-container.sh"
```
